pkg/controllers/networkpolicy/policy_controller.go

```diff
@@ -15,126 +15,76 @@
-			Get().
-			Resource("networkpolicies").
-			Timeout(10 * time.Second).
-			Do().Into(&list)
-		return &list, err
-	}
+	listWatcher := cache.NewListWatchFromClient(clientset.Extensions().RESTClient(), "networkpolicies", "", fields.Everything())
 
 	// Function returns map of policyName:policy stored by policy controller
 	// in datastore.
 	listFunc := func() (map[string]interface{}, error) {
-		var policyMap map[string]bool
-
 		// Get all policies from datastore
-		calicoPolicies, err := calicoClient.Policies().List(api.PolicyMetadata{})
+		calicoPolicies, err := c.NetworkPolicies().List(context.Background(), options.ListOptions{})
 		if err != nil {
 			return nil, err
 		}
 
 		// Filter in only objects that are written by policy controller.
 		m := make(map[string]interface{})
 		for _, policy := range calicoPolicies.Items {
-			policyName := policyConverter.GetKey(policy)
-			if strings.HasPrefix(policyName, "knp.default.") {
-				m[policyName] = policy
-			} else if len(strings.Split(policyName, ".")) > 1 {
-				// Older versions of the controller used the name format `namespace.name`.
-				// This is a best-effort attempt to sync those policies as well.  If it has a `.` and
-				// exists in the k8s API, then sync the policy.
-				// TODO: Remove this section once we don't care about upgrade from the Python controller.
-				l, err := listWatcher.List(metav1.ListOptions{})
-				if err != nil {
-					log.WithError(err).Warnf("Failed to process policy %s", policyName)
-				}
-				items, err := meta.ExtractList(l)
-				if err != nil {
-					log.WithError(err).Warnf("Failed to process policy %s", policyName)
-				}
-				if policyMap == nil {
-					// Populate the policy map with data from the API so we can do a lookup
-					// on whether or not this policy exists in the k8s API.  Only do this if we haven't
-					// already populated the map in order to minimize work.
-					policyMap = map[string]bool{}
-					for _, i := range items {
-						k := fmt.Sprintf("%s.%s", i.(*extensions.NetworkPolicy).Namespace, i.(*extensions.NetworkPolicy).Name)
-						policyMap[k] = true
-					}
-				}
-				log.Debugf("Checking if we care about policy %s", policyName)
-				log.Debugf("Policies we might care about: %#v", policyMap)
-				if _, ok := policyMap[policyName]; ok {
-					// The policy exists in the API and in etcd - assume it was created by the old policy
-					// controller and add it to the batch so that we sync it away.
-					log.Infof("Assuming we're responsible for policy %s", policyName)
-					m[policyName] = policy
-				}
-			} else if policyName == "k8s-policy-no-match" {
-				// Older versions of the controller programmed this policy, but we don't
-				// want it around any more.  TODO: Remove this section once we don't care about
-				// upgrade from the Python controller.
-				log.Infof("Assuming we're responsible for policy %s", policyName)
-				m[policyName] = policy
+			if strings.HasPrefix(policy.Name, "knp.default.") {
```

Comment:
minor suggestion: we might want to make `knp.default.` a global constant.

Reply:
Yeah, going to leave as is for now. It's only used here.